### ISSUE TITLE
Doesn't work at all with Zotero 5.0.97

### ISSUE THREAD

**Issue by aidan-gibson** [PubPeerFoundation/pubpeer_zotero_plugin#25]

**Comment by retorquere** [comment]
@brandonStell I've created a [5-0-97 branch](https://github.com/PubPeerFoundation/pubpeer_zotero_plugin/tree/zotero-5-0-97) that incorporates the necessary changes that makes the pubpeer plugin compatible with both 5.0.96.3 and 5.0.97. The incompatibility was announced [here](https://forums.zotero.org/discussion/91449/add-ons-broken-in-the-latest-beta-version-bbt-zutilo-etc#latest), and the requisite changes were announced [here](https://github.com/scitedotai/scite-zotero-plugin/pull/26), but it took most plugin devs by surprise.

This new branch is untested, but the changes were minor in the end; you can see what changed by checkout out the branch and running `git diff master`.

I've also replaced `webpack` with `esbuild` -- sorry for not discussing, this, but for compatibility with both 5.0.96.3 and 5.0.97 I needed to exclude a `require` call from being usurped by webpack, and webpack has become unmanageably complex. Changing to esbuild was simpeler than figuring out webpack. esbuild is also massively faster.

**Comment by retorquere** [comment]
That is unchanged, just run `npm run build`. You'll have to run `npm i` once before that though, I updated the npm requirements.

**Comment by aidan-gibson** [comment]
Still no dice,unfortunately. I cloned that branch, used `npm i` and `npm run build` which generated the .xpi, but PubPeer doesn't work. I see it in the context menu "Check for comments on pubpeer" but no data propegates on the right side of zotero. If I can provide any further data please let me know.

**Comment by retorquere** [comment]
The change was more meant to give the developers the start of Zotero 5.0.97 compt rather than a working fix (which is why I've committed on a branch rather than committing on master, but I'll see if it's an easy fix.

**Comment by retorquere** [comment]
The current state on the branch should work. I've done only minimal testing.

**Comment by champagnealexandre** [comment]
@retorquere thanks for the fix ; will it be committed on master soon? I’m currently running Zotero 5.0.97-beta.57+07df7d0de, the one on the dev build page on zotero.org

**Comment by retorquere** [comment]
The fix is already on master (and the 5-0-97 branch no longer exists); if you create a new version, a new release will be published automatically.

**Comment by champagnealexandre** [comment]
I’m sorry I don’t understand that last part of your message — creating a new version?

**Comment by retorquere** [comment]
If you issue `npm version <level>` in a clone of the repo , where level is `major`, `minor` or `patch` (see semver.org for what those levels mean) it will bump up the version number, and publish that version. After publishing, zotero of your users will start upgrading within 48 hours IIRC.

**Comment by champagnealexandre** [comment]
> zotero of your users will start upgrading

still don’t understand what you mean, I don’t "have" any users ; I also don’t have any clone of the repo, I downloaded the dmg of the beta from zotero.org

by "committed on master soon" I meant will this be integrated in the publicly available dmg on zotero.org

**Comment by retorquere** [comment]
Oh sorry, I thought you were involved with development of this plugin. Then the answer to your question is "it is already committed on master"

**Comment by champagnealexandre** [comment]
No worries. So that means when the next updated dmg is available on zotero.org the fix will be included and the plugin will work?

**Comment by retorquere** [comment]
I don't follow? This isn't a fix to Zotero.

**Comment by champagnealexandre** [comment]
I thought that when you said "I've created a 5-0-97 branch" (first message of this issue) you were talking about a _Zotero_ branch (because of the similarity in version number). I just came here because the plugin didn’t work — I’m a user, not part of any dev team related to Zotero or its plugins. That’s why I was wondering if there was a fix coming in for the end user. Sorry for the confusion.
